### ISSUE TITLE
cms_banners

### ISSUE THREAD

**Issue by Kaan2106** [Sonykko/Havana_theallseeingeye#4]
I just wanted to visite the ASE/HK on submenu "Campaign Management" / "Ads banners".
the site just kept loading into infinite, so I checked the db and figured out there isnt any entry yet.

I just slapt one entry into it, using localhost as adress and then the webserver crashes

This is the crash-log from today: https://pastebin.com/nyU8nuyj - pas: X2QjSHQmNU

**Comment by Kaan2106** [comment]
By simply removing my hand-inserted "cms_banners" entry in the database the error stopped to be happening again

**Comment by Sonykko** [comment]
Hello! Thanks for notice it. Don't add manually the entry from DB, use the tool cause is designed to insert the correct data to it. I updated the SQL for add a default entry for cms_banners, whatever this is the SQL that you need: 

```
INSERT INTO `cms_banners` (`id`, `text`, `banner`, `url`, `status`, `advanced`, `order_id`) VALUES
(1, 'Keep it (un)real!', 'https://i.imgur.com/6JN4Vwu.gif', 'https://forum.ragezone.com/members/861739.html', '1', '0', 1);
```

And be carefoul, you can add a advanced or not banner, two options have a diferent structure, that is explained in the tool inside Housekeeping. If is advanced banner, you only need to add all your HTML inside Banner input (and complete status, set is advanced one and the order ID), if not, you need to do all inputs except the Banner one, and set as is not advanced one.

Tell me if this solves your issue!

**Comment by Kaan2106** [comment]
Indeed, this solved my issue - thank you.

Whats the difference between the normal banner and a advanced one?

**Comment by Sonykko** [comment]
Perfect! U welcome!

How already I explained, both have a diferent structure.

The normal (not advanced) banner the information of it is splitted btween the inputs of the tool (URL, text, banner image URL, etc...). That mean have a predefined structure, that creates the banner HTML with all the inputs you send.

The advanced one the information is not splitted. Directly you only need to put your own HTML inside Banner input. That modality is more flexible for customize your banner. And it's only need to do the Banner input and the configuration inputs (is advanced, is hidden/not hidden and order ID).

For example, if you want to do a one banner in not advanced mode you create this SQL with the tool (yes, the same SQL that I posted up):

````
INSERT INTO `cms_banners` (`id`, `text`, `banner`, `url`, `status`, `advanced`, `order_id`) VALUES
(1, 'Keep it (un)real!', 'https://i.imgur.com/6JN4Vwu.gif', 'https://forum.ragezone.com/members/861739.html', '1', '0', 1);
````

But if you want to do like a advanced banner the SQL that creates with the tool is this:

````
INSERT INTO `cms_banners` (`id`, `text`, `banner`, `url`, `status`, `advanced`, `order_id`) VALUES
(1, '', '<a href="https://forum.ragezone.com/members/861739.html"><img src="https://i.imgur.com/6JN4Vwu.gif"><p>Keep it (un)real!</p></a>', '', '1', '0', 1);
````

Basically the advanced banner allows you to customize more your banner allowing a custom structure too cause you put directly your HTML, and the not advanced one is more quickly and easy to do, but less customizable and with a predefined structure cause it auto generates the HTML.

**Comment by Kaan2106** [comment]
Thank you very much, I appreciate your fast answers <3

I noticed, that editing a banner isn't possible due to an error 
"Please enter a valid Banner order ID" - I just clicked on edit the 1 banner and wanted to disable it. 
![Screenshot 2024-09-07 194107](https://github.com/user-attachments/assets/7f79bcde-1ade-49d1-bc7b-5d610e8cf299)


There is also an error in showing the advanced banner you provided on the front-end
![Screenshot 2024-09-07 194241](https://github.com/user-attachments/assets/a36e8909-5a0a-47e1-a608-7c7044331728)

Its html is rendered as followed:
`<div class="ad-container">
<a target="_blank" href=""><img src="<a href=" https:="" forum.ragezone.com="" members="" 861739.html"=""><img src="https://i.imgur.com/6JN4Vwu.gif"><p>Keep it (un)real!</p></a>"&gt;<br><a target="_blank" href=""></a><br></div>`

I hope I don't disturb your day 👋

**Comment by Sonykko** [comment]
:)

And yes, I mistake the SQL code for advanced one 🤣 

The correct SQL is it:

````
(1, '', '<a href=\"https://forum.ragezone.com/members/861739.html\">      \r\n <img src=\"https://i.imgur.com/6JN4Vwu.gif\">\r\n <p>Keep it (un)real!</p>\r\n</a>', '', '1', '1', 1);
````

With this SQL the banner go work properly.

And for the error of Order ID I need to take a look when I arrive to home, cause for some reason now I have the same error.

And no! Don't disturb me, Im here to help too :)

**Comment by Kaan2106** [comment]
Oh wow, yeah this query works like a charm 😄, no error at front-end. 

I'll keep you informed if I found something about that edit-error 👍

**Comment by Sonykko** [comment]
Perfect!!! 😄 

Now the back-end is fixed. I pushed a new release with these changes. Now works properly the Ads banners feature.

And thank's for report the bugs, cause it makes improve the project ❤️ 

Tell mi if now you get working well the Ads banners feature!

Download the latest release here: [https://github.com/Sonykko/Havana_theallseeingeye/releases](https://github.com/Sonykko/Havana_theallseeingeye/releases)

**Comment by Kaan2106** [comment]
I noticed that I can't access any of these links inside the tab "system status" - every link gives me this screen
![Screenshot 2024-09-07 215624](https://github.com/user-attachments/assets/9a725e24-138a-4717-baa1-eed1ff446d11)

**Comment by Sonykko** [comment]
Okay, let's see if in the last version you get work all properly. And this error is the first time I see, I so strange. Make sure you run the theallseeingeye.sql in the correct order. And check if the settings table has added the columnn category and added the settings_desc table too. And tell me in what moment you have this error, if always have it or is a new error for you.

**Comment by Kaan2106** [comment]
Just reinstalled the whole project for about 3 times and I can't access configuration settings.
• inside hklogs.action.tpl is a typo: `+{% include "housekeeping/base/header.tpl" %}` (+ needs to be removed)
• found some wired bug at the room badges section
![Screenshot 2024-09-08 015928](https://github.com/user-attachments/assets/43f969f2-9523-404a-ae53-60c9292f94bb)
• staff picks: by submitting the form (without filling anything in, you'll get a 404 error view of front-end instead of error message)
also in staffpicks: the front-end tells me a picked room from alex with id 1, but there is no entry to that (according to staff picks page). I've looked at every table in db, can't find Alex's picked room 
![Screenshot 2024-09-08 021445](https://github.com/user-attachments/assets/a89cc700-fb01-4b25-b6d6-fa6f16d09eb5)

I don't want to spam this thread since the banner issue got fixed asap ❤️  Thank you again 😄

**Comment by Sonykko** [comment]
Just check the things that I mentioned in my last comment about the settings. I try the project again (Github source) in a fresh installation and the System status/Config settings work properly for me, so check too if you followed all steps in correct order of installation guide.

-hklogs_action.tpl has been removed the unused code (+ sign).
-room_badges.tpl has been removed the unused code (duplicated footer).
-Has been fixed Staff Picks tool. Download the latest release and you get work properly :)

And for the Rooms Staff Picks just like is mentioned in the Features list, at the moment are missing the rooms feature cause Quackster didn't code it.

Ye hahaha if found more bugs or have any question open a new issue/pull request, here let's finish the things that are spoken and when got finished I go close the issue 👍

**Comment by Kaan2106** [comment]
Yeah I checked the database, all the neccessary entries were added - as here, both settings and settings_desc were there and are filled in, but getting the same HK error in system_status/configurations as mentioned above 😄 
![Screenshot 2024-09-08 204921](https://github.com/user-attachments/assets/a04af204-fafa-468b-a920-f939cdae9aab)

**Comment by Sonykko** [comment]
You didn't follow the installation guide correctly juts like i said, see how you have empty the category column in settings table. And I hope you get working well the Staff Picks feature 👍

**Comment by Kaan2106** [comment]
I'll setup a new database and try to duplicate my error / mistake, firstly havana.sql, then groups.sql, after that theallseeingeye.sql is the correct order?

(Is it required to run the cleanup afterwards?)

I keep you updated, if its gone im gonna close with comment 👍

**Comment by Sonykko** [comment]
Just follow the installation guide steps in the correct order again. Check it here: [https://github.com/Sonykko/Havana_theallseeingeye/tree/develop?tab=readme-ov-file#installation](https://github.com/Sonykko/Havana_theallseeingeye/tree/develop?tab=readme-ov-file#installation)

**Comment by Kaan2106** [comment]
completed as the server NEEDS to be running after the second step - the server will make entries into the database.
